### ISSUE TITLE
Llama 4: eager attention results in wrong casual mask shape

### ISSUE THREAD

**Issue by Qubitium** [huggingface/transformers#37322]
### System Info

Latest transformers 4.51.0

### Who can help?

@ArthurZucker @SunMarc  @MekkCyber  I know you guys are probably burning an all-nighter due to Llama 4. =)

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

While adding GPTQModel support Llama 4 which does transformers inference at the layer/module level, we found issues where when eager attention is used, casual mask shapes are off by exactly factor of 2 or 4 for Llama 4:

https://github.com/ModelCloud/GPTQModel/pull/1508

![Image](https://github.com/user-attachments/assets/882c784c-40cc-41b5-a9c4-5e0611cc6c7c)

![Image](https://github.com/user-attachments/assets/7c64121c-3438-406a-86e2-21dd71eeef70)

We are able to bypass this for now by setting `batch=1`, removing padding, and not passing `attention_mask` for inference. 

### Expected behavior

No error.

**Comment by ArthurZucker** [comment]
Thanks will have a look asap!

**Comment by ArthurZucker** [comment]
I think this is fixed, closing
